x-pack/plugins/elastic_assistant/server/routes/utils.ts

```diff
@@ -189,3 +194,26 @@ export const getLlmClass = (llmType?: string, bedrockChatEnabled?: boolean) =>
     : llmType === 'gemini' && bedrockChatEnabled
     ? ActionsClientChatVertexAI
     : ActionsClientSimpleChatModel;
+
+export const isOpenSourceModel = (connector?: Connector): boolean => {
+  if (connector == null) {
+    return false;
+  }
+
+  const llmType = getLlmType(connector.actionTypeId);
+  const connectorApiUrl = connector.config?.apiUrl
+    ? (connector.config.apiUrl as string)
+    : undefined;
+  const connectorApiProvider = connector.config?.apiProvider
+    ? (connector.config?.apiProvider as OpenAiProviderType)
+    : undefined;
+
+  const isOpeAiType = llmType === 'openai';
+  const isOpenAI =
+    isOpeAiType &&
+    (!connectorApiUrl ||
+      connectorApiUrl === OPENAI_CHAT_URL ||
+      connectorApiProvider === OpenAiProviderType.AzureAi);
```

Comment:
Did you consider adding a new `apiProvider` of `Open Source` and relying on that for the `isOpenSourceModel` determination?

Reply:
Yes, my second PR is about that https://github.com/elastic/kibana/pull/194831

We will add another provide to cover "other" OpenAI compatible services. It will work for all newly added connectors, though we would like to support already created connectors - OSS models via OpenAI provide.